src/configure/templateInputHelper/InputControl.ts

```diff
@@ -123,6 +173,92 @@ export class InputControl {
         return inputs;
     }
 
+    private getStaticValidationsResult(value: string): string {
+        let validationResult = "";
+        if (this.inputDescriptor.isRequired) {
+            let result = StaticValidator.validateRequired(value);
+            if (result) {
+                validationResult = validationResult.concat(result);
+            }
+        }
+        if (this.inputDescriptor.staticValidation) {
+            if (!!this.inputDescriptor.staticValidation.minLength || !!this.inputDescriptor.staticValidation.maxLength) {
+                let result = StaticValidator.validateLength(value, this.inputDescriptor.staticValidation.minLength, this.inputDescriptor.staticValidation.maxLength);
+                if (result) {
+                    validationResult = validationResult.concat(result);
+                }
+            }
+            if (!!this.inputDescriptor.staticValidation.pattern) {
+                var regexPattern = this.inputDescriptor.staticValidation.pattern;
+
+                let result = StaticValidator.validateRegex(value, regexPattern, this.inputDescriptor.staticValidation.regexFlags || "");
+                if (result) {
+                    validationResult = this.inputDescriptor.staticValidation.errorMessage ? validationResult : validationResult.concat(result);
+                }
+            }
+            if (this.inputDescriptor.type === InputDataType.Int && (!!this.inputDescriptor.staticValidation.minValue || !!this.inputDescriptor.staticValidation.maxValue)) {
+                let result = StaticValidator.validateNumberValue(value, this.inputDescriptor.staticValidation.minValue, this.inputDescriptor.staticValidation.maxValue);
+                if (result) {
+                    validationResult = validationResult.concat(result);
+                }
+            }
+        }
+        return validationResult;
+    }
+
+    private async evaluateDynamicValidation(validation: InputDynamicValidation, value: string): Promise<string> {
+        var requiredDataSource: DataSource = null;
+        this.validationDataSourceToInputsMap.forEach((descriptorsArray, dataSource) => { if (dataSource.id === validation.dataSourceId) { requiredDataSource = dataSource; } });
```

Comment:
add types for descriptorsArray and dataSource

Reply:
done